### PR TITLE
Nanny/Worker with unspecified protocol will get from scheduler address

### PR DESCRIPTION
**Pull request by pentschev**
This ensures protocol is taken from the scheduler address when unspecified. If the user doesn't pass it when using a different protocol (such as `ucx`), it will cause the worker to use `tcp` by default and make communication not work as intended. Therefore, it's safer to get from the scheduler address to prevent faulty behavior that can be difficult to detect.

cc @mrocklin

### CODE REVIEW

**Comment by pentschev** [comment]
I'm still thinking of a suitable test, but haven't yet figured one without making `protocol` an attribute of `Nanny`/`Worker` that I can use to assert later on. Suggestions welcome.

**Comment by mrocklin** [comment]
Maybe make a scheduler with UCX and then verify that `worker.address.startswith("ucx://")` ?

**Comment by pentschev** [comment]
> Maybe make a scheduler with UCX and then verify that `worker.address.startswith("ucx://")` ?

But that would not run on CI today since AFAIK there's no UCX available. Is that still ok?

**Comment by mrocklin** [comment]
It's fine with me

**Comment by pentschev** [comment]
@mrocklin I've added such test. As discussed before, CI isn't going to run it, and to run locally one needs #3177 as well.

File: distributed/tests/test_worker.py [hunk 1]
```diff
@@ -1497,6 +1497,22 @@ def test_resource_limit(monkeypatch):
                 assert all("127.0.0.1" == d["host"] for d in info["workers"].values())
 
 
+@pytest.mark.asyncio
+@pytest.mark.parametrize("Worker", [Worker, Nanny])
+@pytest.mark.parametrize("protocol", ["tcp", "ucx"])
```
Comment (mrocklin) [inline distributed/tests/test_worker.py]:
```suggestion
@pytest.mark.parametrize("protocol", ["ucx"])
```

I recommend that we remove tcp here to keep tests somewhat fast.

**Comment by mrocklin** [comment]
Thanks @pentschev !

**Comment by pentschev** [comment]
Thanks for reviewing and merging @mrocklin !

**Pull request merged**